pusher/src/Controller/AuthenticateController.ts

```diff
@@ -48,21 +49,62 @@ export class AuthenticateController extends BaseController {
             res.onAborted(() => {
                 console.warn("/message request was aborted");
             });
-            const { code, nonce } = parse(req.getQuery());
+            const { code, nonce, token } = parse(req.getQuery());
             try {
+                //verify connected by token
+                if (token != undefined) {
+                    try {
+                        const authTokenData: AuthTokenData = jwtTokenManager.verifyJWTToken(token as string, false);
+                        if (authTokenData.hydraAccessToken == undefined) {
+                            throw Error("Token cannot to be check on Hydra");
+                        }
+                        await openIDClient.checkTokenAuth(authTokenData.hydraAccessToken);
+                        res.writeStatus("200");
+                        this.addCorsHeaders(res);
+                        return res.end(JSON.stringify({ authToken: token }));
+                    } catch (err) {
+                        console.info("User was not connected", err);
+                    }
+                }
+
+                //user have not token created, check data on hydra and create token
                 const userInfo = await openIDClient.getUserInfo(code as string, nonce as string);
                 const email = userInfo.email || userInfo.sub;
                 if (!email) {
                     throw new Error("No email in the response");
                 }
-                const authToken = jwtTokenManager.createAuthToken(email);
+                const authToken = jwtTokenManager.createAuthToken(email, userInfo.access_token);
                 res.writeStatus("200");
                 this.addCorsHeaders(res);
                 return res.end(JSON.stringify({ authToken }));
             } catch (e) {
                 return this.errorToResponse(e, res);
             }
         });
+
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        this.App.get("/logout-callback", async (res: HttpResponse, req: HttpRequest) => {
```

Comment:
why do you need a logout callback? Simply deleting the token is enough

Reply:
Logout callback permit to defined logout user in hydra